common/cpu/amd/kvm.nix

```diff
@@ -0,0 +1,4 @@
+{
+  imports = [ ./. ];
+  boot.kernelModules = [ "kvm-amd" ];
```

Comment:
I think if we include this in nixos-hardware it can go directly into the amd and intel profile. The reason this has been not done so far is that the nixos-generate-config module already will add this to `hardware-configuration.nix`.
So this is the actual question, should we rather add a more comprehensive intel/amd profile to nixos and just enable a flag in both `hardware-configuration.nix` and `nixos-hardware` when included or do we just live with the duplication between the two?

Reply:
I think I might have misunderstood the role of `hardware-configuration.nix` in a "pure flake" NixOS config, and what the goals are of `nixos-hardware`. If others are generating and referencing `hardware-configuration.nix` using `--impure` then I think I confused myself somewhere while reading about using a NixOS flake and made things more difficult than they needed to be.

Reply:
You should not need `--impure` if you are using flakes or tools like niv for importing nixos-hardware in your configuration.
Just now we are trying to supplement things that are not detected by `nixos-generate-config`.